assets/js/components/GoogleChart.js

```diff
@@ -20,205 +20,251 @@
+	if ( onSelect ) {
+		combinedChartEvents.push( {
+			eventName: 'select',
+			callback: onSelect,
+		} );
+	}
 
 	return (
-		<div className="googlesitekit-graph-wrapper">
-			<div ref={ chartRef } className="googlesitekit-line-chart">
-				<div className="googlesitekit-chart-loading">
-					{ loading && (
-						<div className="googlesitekit-chart-loading__wrapper">
-							{ loadText && (
-								<p>{ __( 'Loading chart…', 'google-site-kit' ) }</p>
-							) }
-
-							<ProgressBar
-								className={ className }
-								small={ loadSmall }
-								compress={ loadCompressed }
-								height={ loadHeight }
-							/>
-						</div>
-					) }
-				</div>
-			</div>
+		<div
+			className={ classnames(
+				'googlesitekit-chart',
+				`googlesitekit-chart--${ chartType }`,
+				className
+			) }
+		>
+			<Chart
+				className="googlesitekit-chart__inner"
+				chartEvents={ combinedChartEvents }
+				chartType={ chartType }
+				chartVersion="49"
+				data={ modifiedData }
+				loader={ loader }
+				height={ height }
+				getChartWrapper={ ( chartWrapper, google ) => {
+					// Remove all the event listeners on the old chart before we draw
+					// a new one. Only run this if the old chart and the new chart aren't
+					// the same reference though, otherwise we'll remove existing `onReady`
+					// events and other event listeners, which will cause bugs.
+					if ( chartWrapper !== chartWrapperRef.current ) {
+						// eslint-disable-next-line no-unused-expressions
+						googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current?.getChart() );
+						// eslint-disable-next-line no-unused-expressions
+						googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current );
+					}
+
+					chartWrapperRef.current = chartWrapper;
+					googleRef.current = google;
+
+					if ( getChartWrapper ) {
+						getChartWrapper( chartWrapper, google );
+					}
+				} }
+				width={ width }
+				{ ...otherProps }
+			/>
+			{ children }
 		</div>
 	);
 }
 
 GoogleChart.propTypes = {
-	chartID: PropTypes.string,
-	chartType: PropTypes.oneOf( [ 'area', 'pie', 'line' ] ).isRequired,
 	className: PropTypes.string,
-	data: PropTypes.arrayOf( PropTypes.array ),
-	loadCompressed: PropTypes.bool,
-	loadSmall: PropTypes.bool,
-	loadHeight: PropTypes.number,
+	children: PropTypes.node,
+	chartEvents: PropTypes.arrayOf( PropTypes.shape( {
+		eventName: PropTypes.string,
+		callback: PropTypes.func,
+	} ) ),
+	chartType: PropTypes.oneOf( [
+		'AnnotationChart',
+		'AreaChart',
+		'BarChart',
+		'BubbleChart',
+		'Calendar',
+		'CandlestickChart',
+		'ColumnChart',
+		'ComboChart',
+		'DiffChart',
+		'DonutChart',
+		'Gantt',
+		'Gauge',
+		'GeoChart',
+		'Histogram',
+		'LineChart',
+		'Line',
+		'Bar',
+		'Map',
+		'OrgChart',
+		'PieChart',
+		'Sankey',
+		'ScatterChart',
+		'SteppedAreaChart',
+		'Table',
+		'Timeline',
+		'TreeMap',
+		'WaterfallChart',
+		'WordTree',
+	] ).isRequired,
+	data: PropTypes.array,
```

Comment:
Actually, I tried that and got errors, because sometimes `data` is actually an array of arrays of objects. But:

```js
PropTypes.arrayOf(
	PropTypes.oneOf(
		PropTypes.arrayOf( PropTypes.object ),
		PropTypes.object,
	),
)
```

works, so I've added that.

Reply:
Actually the contents are really varied and the size of the arrays varies too. I don't think there's much benefit to doing anything beyond `array` here:

<img width="448" alt="Screenshot 2021-05-05 at 18 10 44" src="https://user-images.githubusercontent.com/90871/117181527-3c823800-adcd-11eb-8fe4-2ec564877e21.png">

Reply:
:facepalm: you are right. For some reason, I thought that we pass an array of objects received from the `getReport` selector. Looks good to me.